tests/unit/test_charm.py

```diff
@@ -19,8 +20,36 @@ def test_flask_pebble_layer(harness: Harness) -> None:
     assert flask_layer == {
         "override": "replace",
         "summary": "Flask application service",
-        "command": "python3 -m gunicorn --chdir /srv/flask/app app:app -b 0.0.0.0:8000",
+        "command": "python3 -m gunicorn -c /srv/flask/gunicorn.conf.py app:app",
         "startup": "enabled",
         "user": "flask",
         "group": "flask",
     }
+
+
+def test_gunicorn_config(harness: Harness, mock_container_fs: dict[str, str]) -> None:
+    """
+    arrange: start the flask charm and set flask-app container to be ready.
+    act: update gunicorn related charm configurations.
+    assert: gunicorn configuration file inside the flask app container should change accordingly.
+    """
+    harness.begin_with_initial_hooks()
+    harness.set_can_connect(harness.model.unit.containers["flask-app"], True)
+    harness.framework.reemit()
+    harness.update_config({"webserver_workers": 1})
+    assert mock_container_fs.get("/srv/flask/gunicorn.conf.py") == "\n".join(
+        ["bind = ['0.0.0.0:8000']", "chdir = '/srv/flask/app'", "workers = 1"]
```

Comment:
Same as above for the binding address and the port

Reply:
Same as above.